spec/models/manageiq/providers/amazon/cloud_manager/metrics_capture_spec.rb

```diff
@@ -0,0 +1,28 @@
+require "spec_helper"
+
+describe ManageIQ::Providers::Amazon::CloudManager::MetricsCapture do
+  before do
+    _guid, _server, @zone = EvmSpecHelper.create_guid_miq_server_zone
+
+    @ems_amazon = FactoryGirl.create(:ems_amazon, :zone => @zone)
+    @vm = FactoryGirl.build(:vm_perf_amazon, :ext_management_system => @ems_amazon)
+  end
+
+  context "With no EMS defined" do
+    it "#perf_collect_metrics raises an error" do
```

Comment:
Prefer to have the "#perf_collect_metrics" as the outermost context...and then the different cases you are testing within that.